compiler/ras/ILValidationRules.cpp

```diff
@@ -496,3 +496,24 @@ void TR::Validate_ireturnReturnType::validate(TR::Node *node)
          }
       }
    }
+
+/**
+ * Validate_ireturnReturnType (a TR::NodeValidationRule):
+ */
+TR::Validate_axaddEnvironment::Validate_axaddEnvironment(TR::Compilation *comp)
+   : TR::NodeValidationRule(comp, OMR::validate_axaddEnvironment)
+   {
+   }
+
+void TR::Validate_axaddEnvironment::validate(TR::Node *node)
+   {
+   auto opcode = node->getOpCode();
+   if (opcode.getOpCodeValue() == TR::aiadd)
+      {
+      TR::checkILCondition(node, comp()->target().is32Bit(), comp(), "aiadd should not be seen on 64-bit");
+      }
+   if (opcode.getOpCodeValue() == TR::aladd)
+      {
+      TR::checkILCondition(node, comp()->target().is64Bit(), comp(), "aladd should not be seen on 32-bit");
+      }
```

Comment:
Although this is a perfectly valid, I would slightly prefer it if the code was
```suggestion
   if (comp()->target().is32Bit())
      {
      TR::checkILCondition(node, opcode.getOpCodeValue() == TR::aiadd, comp(), "aiadd should not be seen on 64-bit");
      }
   if (comp()->target().is64Bit())
      {
      TR::checkILCondition(node, opcode.getOpCodeValue() == TR::aladd, comp(), "aladd should not be seen on 32-bit");
      }
```

because it's the opcode we use that depends on the target, not the other way around.

Reply:
If changing in this way, other nodes (except for `aiadd` & `aladd`) will fail in the test. Do you think I should put a `if` condition to constrain the codes above? Or if there is any way to only allow `aiadd` & `aladd` to run this verifier?

Reply:
Oh, that's right. I momentarily forgot how the IL validator works. Let's pretend I didn't make that suggestion. :sweat_smile: